src/main/java/org/mitre/synthea/world/concepts/Terminology.java

```diff
@@ -0,0 +1,575 @@
+package org.mitre.synthea.world.concepts;
+
+import ca.uhn.fhir.context.FhirContext;
+import com.google.common.collect.ArrayListMultimap;
+import com.google.common.collect.Multimap;
+import com.google.gson.JsonObject;
+import com.google.gson.JsonParser;
+import com.google.gson.stream.JsonReader;
+
+import java.io.FileNotFoundException;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.io.StringReader;
+import java.net.InetSocketAddress;
+import java.net.Proxy;
+import java.net.URISyntaxException;
+import java.net.URL;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.Random;
+import java.util.concurrent.ConcurrentHashMap;
+
+import javax.xml.parsers.DocumentBuilder;
+import javax.xml.parsers.DocumentBuilderFactory;
+
+import okhttp3.FormBody;
+import okhttp3.HttpUrl;
+import okhttp3.OkHttpClient;
+import okhttp3.Request;
+import okhttp3.RequestBody;
+import okhttp3.Response;
+
+import org.hl7.fhir.dstu3.model.ValueSet;
+import org.mitre.synthea.helpers.Config;
+import org.mitre.synthea.world.concepts.HealthRecord.Code;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.w3c.dom.Document;
+import org.w3c.dom.Element;
+import org.w3c.dom.NamedNodeMap;
+import org.w3c.dom.Node;
+import org.w3c.dom.NodeList;
+import org.xml.sax.InputSource;
+
+
+
+
+
+
+/**
+ * Finds a specified value set, extracts the codes and systems
+ * and selects one code/system at random.
+ */
+
+public class Terminology {
+
+  private static String VS_FOLDER = "valuesets";
+
+
+
+  private static final String VSAC_USER = Config.get("terminology.username");
+  private static final String VSAC_PASS = Config.get("terminology.password");
+
+  private static final String AUTH_URL = "http://vsac.nlm.nih.gov/vsac/ws/Ticket";
+
+  private static final String SERVICE_URL = "http://umlsks.nlm.nih.gov";
+
+  private static final String VSAC_VALUE_SET_RETRIEVAL_URL = "vsac.nlm.nih.gov";
+  private static final FhirContext ctx = FhirContext.forDstu3();
+  private static final Map<String, ValueSet> valueSets = loadValueSets();
+  private static final Map<String,String> codeLookup = loadLookupTable();
+
+  private static final Logger logger = LoggerFactory.getLogger(Terminology.class);
+  private static boolean connectionFailed = false;
+
+
+  public static Session sess;
```

Comment:
This is more personal preference but a more descriptive name here would be nice. This is **a** Session but Session is public so is there a specific reason I would want to use this one vs creating another one?
Also this is public static but not final so there's the possibility that it could change unexpectedly. Maybe it should be final?

Reply:
This will be removed when I address the issue below.